src/main/java/com/google/step/datamanager/DealManager.java

```diff
@@ -21,6 +21,12 @@ public Deal createDeal(
 
   public void deleteDeal(long id);
 
+  public List<Deal> getDealsPublishedByFollowedUsers(List<Long> userIds);
+
+  public List<Deal> getDealsPublishedByFollowedRestaurants(List<Long> restaurantIds);
```

Comment:
Drop the word "Followed". Now this is just taking userIds / restaurantIds and returning deals published by them. It has nothing to do with "following".